### ISSUE TITLE
Visible Over Fog of War objects are always visible

### ISSUE THREAD

**Issue by Phergus** [RPTools/maptool#2151]
**Describe the bug**
With current develop branch, items marked as Visible over Fog of War are always visible.

**To Reproduce**
1. Load attached campaign.
2. Go to Hero token and expose current area.
3. Switch to Player View
4. See all of the VOFoW  doors on map:
![image](https://user-images.githubusercontent.com/34379254/89364781-84363080-d690-11ea-96e3-aa528bc291ad.png)

Same thing is seen on a connected player client.

**Expected behavior**
Expect only doors within visual range of token.

**MapTool Info**
- Version: Current develop branch at least as far back as July 20th.

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: 10

**Additional context**
[DonjonMapTest180alpha.cmpgn.zip](https://github.com/RPTools/maptool/files/5025795/DonjonMapTest180alpha.cmpgn.zip)
Remove `.zip` suffix.

**Comment by ebudai** [comment]
I'll take a stab at this one.

**Comment by Phergus** [comment]
Go for it.   Be aware that there are a lot of complex interactions between sight, light and VBL then compounded by GM vs Player view plus Individual Views and so on.

**Comment by JamzTheMan** [comment]
I know this didn't do this originally, at least for IViews/IFoW. May be worth it to test in 1.5.0 and if it behaves differently, may follow through and see when the bug was introduced.

**Comment by Phergus** [comment]
MT 1.7.0 works correctly so it is something that popped up in develop.  It's present in code back to 7/20 so some time before that.

**Comment by JamzTheMan** [comment]
Ah well that makes it easier then :)
To the Git Blame!

**Comment by nmeier** [comment]
Could be related to #2069

**Comment by nmeier** [comment]
yeah, broken in https://github.com/RPTools/maptool/commit/eeb57c9ea3108631abaa3e3c6930d1bf745e934a

I was tweaking that over-fow tokens would become visible (again) when a certain amount of surface is visible.

**Comment by ebudai** [comment]
I found the same, and have restored the offending bit of code (in timer 7).  I'm trying to reduce it now as there are a lot of DRY violations.

**Comment by nmeier** [comment]
it's a brutal method - the changes in #1801 and #2069 were done to speed up rendering and also refactor the method a bit. There's a test case in #2609 that doesn't cover tokens in the object layer exposed here. 
I think I know how to add support for the object layer case by removing more code that changes the flow for clipping FYI.

**Comment by Phergus** [comment]
@nmeier   want to point out that Visible over Fog of War is limited to the Object layer.  Any token, on any layer can be flagged as VOFoW.

**Comment by Phergus** [comment]
@ebudai  your change breaks everything but Visible Over Fog of War tokens.

**Comment by ebudai** [comment]
It appears an experiment I did got in there somehow.  Apologies for the sloppiness; I have an incoming PR that fixes my breakage.

**Comment by Phergus** [comment]
That seems to have restored functionality. Some more testing to do.

**Comment by Phergus** [comment]
Tested. Other than issue noted in #2208, appears to be working correctly.
